### PR TITLE
feat(ts): add the attributesToTransliterate setting

### PR DESCRIPTION
**Pull request by Haroenv**
fixes https://github.com/algolia/algoliasearch-client-javascript/issues/1243

### CODE REVIEW

**Comment by chloelbn** [comment]
Note: this attribute is only applied when your query language is Japanese

**Review by chloelbn — DISMISSED** [review]
Note: this attribute is only applied when your query language is Japanese

**Comment by Haroenv** [comment]
doesn't "for compatible scripts" imply that already? @chloelbn

**Comment by chloelbn** [comment]
@Haroenv indeed! I didn't get that with the wording sorry.

**Review by chloelbn — APPROVED** [review]

File: packages/client-search/src/types/Settings.ts [hunk 1]
```diff
@@ -291,4 +291,10 @@ export type Settings = {
    * results.
    */
   readonly decompoundQuery?: boolean;
+
+  /**
+   * Specify on which attributes in your index Algolia should apply
+   * transliteration to a normalized form for compatible scripts.
```
Comment (remy-zeiss) [inline packages/client-search/src/types/Settings.ts:297]:
```suggestion
   * Specify on which attributes in your index Algolia Japanese transliteration should apply
   * to make those attributes searchable in Hiragana.
```

Comment (remy-zeiss) [inline packages/client-search/src/types/Settings.ts:297]:
```suggestion
   * Specify on which attributes in your index Algolia Japanese transliteration should apply
   * to make those attributes searchable in Hiragana.
```

Comment (remy-zeiss) [inline packages/client-search/src/types/Settings.ts:297]:
```suggestion
   * Specify on which attributes in your index Algolia should apply
   * Japanese transliteration to make words indexed in Katakana or Kanji searchable in Hiragana.
```

**Pull request merged**